### PR TITLE
Fix URLs in NeptuneLogger docstring

### LINKED ISSUE #21371: Broken URLs on NeptuneLogger documentation page
### 📚 Documentation

Neptune changed their API and docs to version 3.X. This unfortunately breaks the URLs on the [NeptuneLogger](https://lightning.ai/docs/pytorch/stable/api/lightning.pytorch.loggers.neptune.html#module-lightning.pytorch.loggers.neptune) docs page.

The URLs should point to the version 2.X legacy docs pages of Neptune instead. 

cc @ethanwharris @lantiga

### PR DESCRIPTION
**Pull request by aknierim**
## What does this PR do?

Neptune updated their API and docs to version 3.X. This small PR  fixes the URLs in the NeptuneLogger docstring to point to the correct version 2.X legacy docs.

Fixes #21371 

<details>
  <summary><b>Before submitting</b></summary>

- Was this **discussed/agreed** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/Lightning-AI/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- Did you make sure to **update the documentation** with your changes? (if necessary)
- Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes?
- Did you list all the **breaking changes** introduced by this pull request?
- Did you **update the CHANGELOG**? (not for typos, docs, test updates, or minor internal changes/refactors)

</details>

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/Lightning-AI/lightning/wiki/Review-guidelines). In short, see the following bullet-list:

<details>
  <summary>Reviewer checklist</summary>

- [ ] Is this pull request ready for review? (if not, please submit in draft mode)
- [ ] Check that all items from **Before submitting** are resolved
- [ ] Make sure the title is self-explanatory and the description concisely explains the PR
- [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified

</details>

----
📚 Documentation preview 📚: https://pytorch-lightning--21372.org.readthedocs.build/en/21372/

### CODE REVIEW

**Comment by deependujha** [comment]
closing PR. Please [check comment](https://github.com/Lightning-AI/pytorch-lightning/issues/21371#issuecomment-3695043868)
